cellpack/autopack/Analysis.py

```diff
@@ -1042,73 +1051,215 @@ def get_obj_dict(self, input_path):
                         all_objs[obj][seed_key] = {}
                     for ct, dim in enumerate(["x", "y", "z"]):
                         all_objs[obj][seed_key][dim] = positions[:, ct]
+                    sph_pts = self.cartesian_to_sph(positions)
+                    for ct, dim in enumerate(["r", "theta", "phi"]):
+                        all_objs[obj][seed_key][dim] = sph_pts[:, ct]
         self.all_objs = all_objs
         self.all_pos_list = all_pos_list
         return all_objs, all_pos_list
 
+    def get_minimum_expected_distance_from_recipe(self, recipe_data):
+        """
+        Returns 2x the smallest radius of objects in the recipe
+        """
+        minimum_distance = 9999
+        for object_values in recipe_data.get("objects").values():
+            if "radius" in object_values:
+                if 2 * object_values["radius"] < minimum_distance:
+                    minimum_distance = 2 * object_values["radius"]
+        return minimum_distance
+
+    def get_packed_minimum_distance(self, glob_to_distance_files):
+        """
+        Returns the minimum distance between packed objects
+        """
+        minimum_packed_distance = 9999
+        for path_to_distance_file in glob_to_distance_files:
+            distance_values = numpy.genfromtxt(path_to_distance_file)
+            current_minimum_distance = numpy.min(distance_values)
+            if current_minimum_distance < minimum_packed_distance:
+                minimum_packed_distance = current_minimum_distance
+
+        return minimum_packed_distance
+
+    def create_report(self, report_options):
```

Comment:
maybe for another PR, but this could be put in our "Writers" file as a more generic md writer

Reply:
Good idea!